src/pages/index.js

```diff
@@ -0,0 +1,319 @@
+    api.addLike(selectedCardID).catch((err) => console.error(err));
+  } else {
+    api.removeLike(selectedCardID).catch((err) => console.error(err));
+  }
+}
+
+function handleDeleteCard(cardElement, data) {
+  selectedCard = cardElement;
+  selectedCardID = data._id;
+  openModal(deletePostModal);
+}
+
+function handleDeleteSubmit() {
+  const originalText = deletePostBtn.textContent;
+  deletePostBtn.textContent = "Deleting...";
+  api
+    .removeCard(selectedCardID)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deletePostModal);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      deletePostBtn.textContent = originalText;
+    });
+}
+
+// === Event Listeners ===
+// Modal Open Events
+editProfileBtn.addEventListener("click", () => {
+  openModal(profileModal);
+  nameInput.value = profileNameElement.textContent;
+  jobInput.value = profileJobElement.textContent;
+});
+
+newPostBtn.addEventListener("click", () => openModal(newPostModal));
+profileEditImageBtn.addEventListener("click", () =>
+  openModal(profileEditImageModal)
+);
+
+// Modal Close Events
+closeProfileBtn.addEventListener("click", () => closeModal(profileModal));
+newPostCloseBtn.addEventListener("click", () => closeModal(newPostModal));
+deletePostBtn.addEventListener("click", handleDeleteSubmit);
+canelDeletePostBtn.addEventListener("click", () => closeModal(deletePostModal));
+deleteModalCloseBtn.addEventListener("click", () =>
+  closeModal(deletePostModal)
+);
+previewCloseBtn.addEventListener("click", () => closeModal(previewModal));
+profileImageEditCloseBtn.addEventListener("click", ()=> closeModal(profileEditImageModal));
+// Profile Form Submission
+profileFormElement.addEventListener("submit", handleProfileFormSubmit);
+function handleProfileFormSubmit(evt) {
+  evt.preventDefault();
+  const originalText = submitProfileBtn.textContent;
+  submitProfileBtn.textContent = "Saving...";
+  api
+    .editUserInfo({ name: nameInput.value, about: jobInput.value })
+    .then((data) => {
+      profileNameElement.textContent = data.name;
+      profileJobElement.textContent = data.about;
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      submitProfileBtn.textContent = originalText;
+    });
+
+  closeModal(profileModal);
+}
+
+// New Post Submission
+newPostFormElement.addEventListener("submit", handleNewPostSubmit);
+function handleNewPostSubmit(evt) {
+  evt.preventDefault();
+  const originalText = newPostSubmitBtn.textContent;
+  newPostSubmitBtn.textContent = "Saving...";
+  api
+    .postCard({ caption: captionInput.value, link: imageLinkInput.value })
+    .then((data) => {
+      const newCard = getCardElement(data);
+      cardsList.prepend(newCard);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      newPostSubmitBtn.textContent = originalText;
+    });
+  captionInput.value = "";
+  imageLinkInput.value = "";
+  closeModal(newPostModal);
```

Comment:
```js

  captionInput.value = "";
  imageLinkInput.value = "";
  closeModal(newPostModal);
```

These 3 lines should be in `.then` in line 247

You need to **reset** the form only in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened, and the user can try to send the data again to the server without having to type the data into the inputs once more.